target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/11-ath10k-caldata

```diff
@@ -102,7 +102,8 @@ case "$FIRMWARE" in
 		ath10k_patch_mac $(macaddr_add $(mtd_get_mac_binary config 0x10008) +2)
 		;;
 	tplink,re355-v1|\
-	tplink,re450-v1)
+	tplink,re450-v1|\
+	tplink,tl-wr902ac-v1)
```

Comment:
If the address is not in 0x5006 or elsewhere on flash, please calculate based on flash location (mtd_get_mac_binary) instead of eth0.

Reply:
I'll dump the ART and come back to this later, I think there is a reason why @pepe2k didn't use this location. But who knows... ;)

Reply:
...and it turned out, that some example MAC, definitely not even close to that on the label. Same for wmac0.

Reply:
I'd also need to revert to factory and check if the actual offset is -1, as my findings on Flash on "tplink" partition suggest it may be actually +2.

Reply:
And actually, this may be more than one partition, by looking on other tplink-safeloader devices. Archer D60 has more than one partition here, and there is a lot of free space - maybe with compound MTD this space could be reused increase roots_data size.

Reply:
> And actually, this may be more than one partition, by looking on other tplink-safeloader devices. Archer D60 has more than one partition here, and there is a lot of free space - maybe with compound MTD this space could be reused increase roots_data size.

IIRC, as we don't use data from these particular partitions, they were all combined into a single one. I don't think you can re-use them, better keep them as it is.

Reply:
I'll try to check them somehow using vendor firmware, maybe this would be possible without opening the case. As the space for rootfs is constrained on this one, and I found some empty erase blocks, having the flash dump and modded U-boot, I'm dared to at least try.

Reply:
I updated this to be read from flash, but offset is still required. The same one as used in ar71xx is kept, what I found previously was in the user-config partition, where it is modifiable.